src/channels/views/ChannelDetails/ChannelDetails.tsx

```diff
@@ -87,6 +107,56 @@ export const ChannelDetails: React.FC<ChannelDetailsProps> = ({ id }) => {
       }
     });
 
+  const onCompleted = (data: ChannelDelete) => {
+    const errors = data.channelDelete.errors;
+    if (errors.length === 0) {
+      notify({
+        status: "success",
+        text: intl.formatMessage({
+          defaultMessage: "Channel deleted"
+        })
+      });
+      closeModal();
+      navigate(channelsListUrl());
+    } else {
+      errors.map(error =>
+        notify({
+          status: "error",
+          text: getChannelsErrorMessage(error, intl)
+        })
+      );
+    }
+  };
+
+  const [deleteChannel, deleteChannelOpts] = useChannelDeleteMutation({
+    onCompleted
+  });
+
+  const channelsChoices = id
+    ? channelsListData?.data?.channels
+        ?.filter(
+          channel =>
+            channel.id !== id &&
+            channel.currencyCode === data?.channel?.currencyCode
+        )
+        .map(channel => ({
+          label: channel.name,
+          value: channel.id
+        }))
+    : [];
+
+  const handleRemoveConfirm = (targetChannelId?: string) => {
+    if (targetChannelId) {
```

Comment:
Don't think we need and if/else here, since the same action happens in both cases, maybe we can simplify: 
```suggestion
 const data = targetChannelId
    ? { id, input: { targetChannel: targetChannelId } }
    : { id }
 deleteChannel({ variables: data })
```